### PR TITLE
Added scroll snapping to order scroller elements (#202re5h)

### PR DESCRIPTION
**Pull request by azkyakhan**

### CODE REVIEW

File: src/views/Home.vue [hunk 1]
```diff
@@ -179,15 +184,22 @@ ion-card > ion-card-header {
 .scroller-content {
   display: flex;
   overflow-x: scroll;
+  overscroll-behavior-x: contain;
+  scroll-snap-type: x mandatory;
+}
+
+.scroller-card {
```
Comment (dt2patel) [inline src/views/Home.vue]: we don't need this. default value is fine

File: src/views/Home.vue [hunk 2]
```diff
@@ -57,7 +57,8 @@
           </ion-item>
     
           <div class="scroller-content">
-            <div class="scroller-item" v-for="i = 1 in 10" :key="i">  
+            <div class="scroller-card" v-for="i = 1 in 10" :key="i">
```
Comment (dt2patel) [inline src/views/Home.vue]: why do we need an extra container?

**Review by dt2patel — CHANGES_REQUESTED** [review]

File: src/views/Home.vue (hunk 2 above)
Comment (azkyakhan) [reply src/views/Home.vue]: When I went through the concept of scroll-snapping, I explored that scroll snapping is used by setting the `scroll-snap-type` property on a container element and the `scroll-snap-align` property on elements inside it. Here I have a container named `scroller-content` on which I set the `scroll-snap-type` then when I was setting the `scroll-snap-align: start` on child i.e. `scroller-item` here, It was moving itself at start of the container without scrolling. So I went through https://www.apple.com/store to see how they have used scroll snapping and then I followed it.

Comment (dt2patel) [reply src/views/Home.vue]: If you leave scroll snap align to its default value you will not need this extra container.

**Comment by dt2patel** [comment]
No improvements made

File: src/views/Home.vue (hunk 2 above)
Comment (azkyakhan) [reply src/views/Home.vue]: There is no default value of `scroll-snap-align` set on the `scroller-item`. If I am not giving any value to `scroll-snap-align`, scroll snapping is not working. So I tried giving it `start end` value (https://docs.w3cub.com/css/scroll-snap-align) and it is looking like this:

https://user-images.githubusercontent.com/82817616/155300610-1cd5db4c-2d1b-4668-8c99-c92039e68493.mp4

File: src/views/Home.vue [hunk 3]
```diff
@@ -188,18 +184,16 @@ ion-card > ion-card-header {
...
-.scroller-card {
-  scroll-snap-align: start;
-}
-
 .scroller-item{
   flex-shrink: 0;
   position: relative;
   width: max-content;
   transform: translateX(calc(max(var(--page-width), 100vw)/2 - var(--page-width)/2));
+  /* set first block, second inline */
```
Comment (dt2patel) [inline src/views/Home.vue]: I don't understand this comment.

Scroller item should snap align to the center

File: src/views/Home.vue [hunk 4]
```diff
@@ -188,18 +184,16 @@ ion-card > ion-card-header {
...
+  scroll-snap-align: start end;
```
Comment (dt2patel) [inline src/views/Home.vue]:
```suggestion
  scroll-snap-align: center;
```

**Review by dt2patel — CHANGES_REQUESTED** [review]

File: src/views/Home.vue (hunk 3 above)
Comment (azkyakhan) [reply src/views/Home.vue]: Done

File: src/views/Home.vue (hunk 4 above)
Comment (azkyakhan) [reply src/views/Home.vue]: Done

File: src/views/Home.vue (hunk 3 above)
Comment (dt2patel) [reply src/views/Home.vue]: Please be more descriptive with comments on PRs, what was done: did you remove something, did you update the comment?

Comment (azkyakhan) [reply src/views/Home.vue]: I have removed that comment sir and replaced `start end` property with `center`.

File: src/views/Home.vue [hunk 5]
```diff
@@ -179,12 +181,16 @@ ion-card > ion-card-header {
...
   display: flex;
   overflow-x: scroll;
+  overscroll-behavior-x: contain;
+  scroll-snap-type: x mandatory;
 }
 
 .scroller-item{
   flex-shrink: 0;
   position: relative;
+  width: max-content;
```
Comment (dt2patel) [inline src/views/Home.vue]: why do you need this? what happens if you don't do this?

**Review by dt2patel — CHANGES_REQUESTED** [review]

File: src/views/Home.vue (hunk 3 above)
Comment (dt2patel) [reply src/views/Home.vue]: thanks for the detailed comment

File: src/views/Home.vue (hunk 5 above)
Comment (azkyakhan) [reply src/views/Home.vue]: I was trying a few things and by mistake I forgot to remove this line. It is of no use now and I have removed it.

**Review by dt2patel — APPROVED** [review]

**Pull request merged**